### PR TITLE
#3305 File completion on remote workspace duplicates directory entries

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
Also fixes 14.0 references to VS SDK. Not sure how 15.0 references leaked into them.

### CODE REVIEW

File: src/Package/Impl/Imaging/ImagesProvider.cs [hunk 1]
```diff
@@ -21,8 +21,14 @@ namespace Microsoft.VisualStudio.R.Package.Imaging {
...
     internal sealed class ImagesProvider : IImagesProvider {
-        private Dictionary<string, ImageMoniker> _monikerCache = new Dictionary<string, ImageMoniker>();
-        private Lazy<Dictionary<string, string>> _fileExtensionCache = Lazy.Create(() => CreateExtensionCache());
+        private readonly Dictionary<string, ImageMoniker> _monikerCache = new Dictionary<string, ImageMoniker>();
+        private readonly Lazy<Dictionary<string, string>> _fileExtensionCache = Lazy.Create(() => CreateExtensionCache());
+
+        private static IVsImageService2 _imageService;
+
+        public ImagesProvider() {
+            _imageService = _imageService ?? VsAppShell.Current.GetGlobalService<IVsImageService2>(typeof(SVsImageService));
```
Comment (MikhailArkhipov) [inline src/Package/Impl/Imaging/ImagesProvider.cs]: This must be main thread. I can try `DispatchOnUIThread` if MEF might call this async.

Comment (AlexanderSher) [inline src/Package/Impl/Imaging/ImagesProvider.cs]: Why have you moved it from the `GetIconForImageMoniker`?

Comment (MikhailArkhipov) [inline src/Package/Impl/Imaging/ImagesProvider.cs]: Because request for icons in remote file completion case comes from a background thread

Comment (MikhailArkhipov) [inline src/Package/Impl/Imaging/ImagesProvider.cs]: Technically we may want to switch to `ConcurrentDictionary` here although probability of concurrent requests is very low.

Comment (AlexanderSher) [inline src/Package/Impl/Imaging/ImagesProvider.cs]: `ConcurrentDictionary` has overhead only on write operations, so if we don't make thousands of writes here, I think we can switch.

Comment (MikhailArkhipov) [inline src/Package/Impl/Imaging/ImagesProvider.cs]: Yep, done.

**Comment by AlexanderSher** [comment]
👍

**Comment by MikhailArkhipov** [comment]
Ouch, need another - @int19h  ?

**Pull request merged**